### PR TITLE
Fix LyShine Entity Reference Crash

### LINKED ISSUE #14344: LyShine Crash: Null Ref Crash Selecting Ui Entity Reference
Editor crashes whenever selecting an entity reference in LyShine.

**Steps to reproduce**
1. Open UI LyShine Editor
2. Select an entity with a UI Component which references another component
3. Double click the entity reference in order to select the entity in the UI canvas
image

**Expected behavior**
Entity which was references is now selected and its component editor is seen

**Actual behavior**
Editor null-ref crash trying to access a non-existent TransformInterface. Ui entities have a specific UiTransformInterface.
 
image

image

**Found in Branch**
Development

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**

### PR DESCRIPTION
**Pull request by AMZN-Gene**
Stop LyShine editor crash which was caused by the editor attempting to access a null transform component.
Adding null checks, but also stopping the editor before it even gets to that point by stopping the Editor from trying to view ui entities in the viewport
Fixes #14344 
image

## How was this PR tested?
Selected entity references in Ui Canvas as well as the normal Editor and made sure everything still worked. Ui entities were selected without crashing, and normal editor would still point to look at the entity

### CODE REVIEW

**Review by lsemp3d — APPROVED** [review]

File: Code/Editor/Plugins/ComponentEntityEditorPlugin/SandboxIntegration.cpp [hunk 1]
```diff
@@ -1116,18 +1116,29 @@ bool SandboxIntegrationManager::CanGoToSelectedEntitiesInViewports()
...
 {
+    AZ::Entity* entity = nullptr;
+    AZ::ComponentApplicationBus::BroadcastResult(entity, &AZ::ComponentApplicationRequests::FindEntity, entityId);
+    if (entity == nullptr)
+    {
+        return false;
+    }
+
+    // Skip if this is a UI component; UI components use LyShine's "UI Canvas" and are not visible in the Editor viewport
+    const AZ::Component* uiElementComponent = entity->FindComponent(AZ::Uuid("{4A97D63E-CE7A-45B6-AAE4-102DB4334688}"));
```
Comment (lsemp3d) [inline Code/Editor/Plugins/ComponentEntityEditorPlugin/SandboxIntegration.cpp:1127]: Consider using a named variable for that Uuid:
`constexpr AZ::Uuid uiCanvasUUID = AZ::Uuid("{4A97D63E-CE7A-45B6-AAE4-102DB4334688}"));`

Comment (AMZN-Gene) [reply Code/Editor/Plugins/ComponentEntityEditorPlugin/SandboxIntegration.cpp]: Thanks, fixed!

**Review by gadams3 — APPROVED** [review]

**Comment by AMZN-alexpete** [comment]
Huh this whole time I thought all[ LyShine entities were in a different EntityContext](https://github.com/aws-lumberyard-dev/o3de/blob/development/Gems/LyShine/Code/Include/LyShine/UiEntityContext.h#L35)

**Comment by AMZN-Gene** [comment]
> Huh this whole time I thought all[ LyShine entities were in a different EntityContext](https://github.com/aws-lumberyard-dev/o3de/blob/development/Gems/LyShine/Code/Include/LyShine/UiEntityContext.h#L35)
:o

File: Code/Editor/Plugins/ComponentEntityEditorPlugin/SandboxIntegration.cpp [hunk 2]
```diff
@@ -1116,18 +1116,30 @@ bool SandboxIntegrationManager::CanGoToSelectedEntitiesInViewports()
...
+        return false;
+    }
+
+    // Skip if this is a UI component; UI components use LyShine's "UI Canvas" and are not visible in the Editor viewport
+    constexpr AZ::Uuid uiElementComponentUuid{ "{4A97D63E-CE7A-45B6-AAE4-102DB4334688}" };
+    const AZ::Component* uiElementComponent = entity->FindComponent(uiElementComponentUuid);
+    if (uiElementComponent != nullptr)
+    {
+        return false;
+    }
```
Comment (mbalfour-amzn) [inline Code/Editor/Plugins/ComponentEntityEditorPlugin/SandboxIntegration.cpp:1132]: Is there any other way to detect and approach this problem that doesn't introduce a hidden dependency on LyShine code here by hard-coding the UI Element UUID type? For example, based on the description of the PR, could/should this look for missing Transforms as opposed to the presence of UI elements?

**Review by AMZN-alexpete — APPROVED** [review]

File: Code/Editor/Plugins/ComponentEntityEditorPlugin/SandboxIntegration.cpp (hunk 2 above)
Comment (AMZN-Gene) [reply Code/Editor/Plugins/ComponentEntityEditorPlugin/SandboxIntegration.cpp]: I like it! Fixed!

**Review by mbalfour-amzn — APPROVED** [review]

**Review by AMZN-Phil — APPROVED** [review]

**Review by cgalvan — APPROVED** [review]

**Review by michabr — APPROVED** [review]

**Comment by michabr** [comment]
> Huh this whole time I thought all[ LyShine entities were in a different EntityContext](https://github.com/aws-lumberyard-dev/o3de/blob/development/Gems/LyShine/Code/Include/LyShine/UiEntityContext.h#L35)

Yes, each UI canvas has its own entity context.

**Comment by AMZN-Gene** [comment]
> How difficult would it be to add a regression test that fails without this change?

Good idea! I'm going to defer this to a new PR since I have some higher priority items and there aren't any existing unit tests in this domain so it'll require some standup #14395

**Pull request merged**